pandas/tests/frame/test_constructors.py

```diff
@@ -807,7 +807,7 @@ def test_constructor_corner_shape(self):
 
     @pytest.mark.parametrize("data, index, columns, dtype, expected", [
         (None, lrange(10), ['a', 'b'], object, np.object_),
-        (None, None, ['a', 'b'], 'int64', np.dtype('int64')),
+        (None, None, ['a', 'b'], 'int64', np.dtype('float64')),
```

Comment:
cc @TomAugspurger this makes it pass, but this looks slightly suspect, e.g. in master

```
In [4]: pd.DataFrame(columns=list('ab'),dtype=int).dtypes
Out[4]: 
a    int64
b    int64
dtype: object
```

though we coerce almost always on assignment

```
In [9]: df = pd.DataFrame(columns=list('ab'),dtype=int)

In [10]: df.loc[0] = 5

In [11]: df
Out[11]: 
   a  b
0  5  5

In [12]: df.dtypes
Out[12]: 
a    int64
b    int64
dtype: object

In [13]: df = pd.DataFrame(columns=list('ab'),dtype=int)

In [14]: df.loc[0, 'a'] = 5

In [15]: df
Out[15]: 
     a   b
0  5.0 NaN

In [16]: df.dtypes
Out[16]: 
a    float64
b    float64
dtype: object

```